src/info/manifest.c

```diff
@@ -381,6 +381,209 @@ manifestNewInternal(void)
+            if (strBeginsWith(path, linkCheck->basePath))
+            {
+                THROW_FMT(
+                    LinkDestinationError,
+                    "link '%s' destination '%s' is in PGDATA",
+                    strZ(manifestPathPg(target1->name)), strZ(manifestTargetPath(this, target1)));
+            }
+
+            // Check if the link destination path already exists
+            const ManifestLinkCheckItem *const link = lstFind(linkCheck->linkList, &path);
+
+            if (link != NULL)
+            {
+                // If both links are files make sure they don't link to the same file
+                if (target1->file != NULL && link->file != NULL)
+                {
+                    if (strEq(target1->file, link->file))
+                    {
+                        const ManifestTarget *const target2 = manifestTarget(this, link->targetIdx);
+
+                        THROW_FMT(
+                            LinkDestinationError,
+                            "link '%s' (%s/%s) destination is the same file as link '%s' (%s/%s)",
+                            strZ(manifestPathPg(target1->name)), strZ(manifestTargetPath(this, target1)), strZ(target1->file),
+                            strZ(manifestPathPg(target2->name)), strZ(manifestTargetPath(this, target2)), strZ(target2->file));
+                    }
+                }
+                // Else error because one of the links is a path and cannot link to the same path as another file/path link
+                else
+                {
+                    const ManifestTarget *const target2 = manifestTarget(this, link->targetIdx);
+
+                    THROW_FMT(
+                        LinkDestinationError,
+                        "link '%s' (%s) destination is the same directory as link '%s' (%s)",
+                        strZ(manifestPathPg(target1->name)), strZ(manifestTargetPath(this, target1)),
+                        strZ(manifestPathPg(target2->name)), strZ(manifestTargetPath(this, target2)));
+                }
+            }
+            // Else add to the link list and check against other links
+            else
+            {
+                // Add the link destination path and sort
+                lstAdd(linkCheck->linkList, &(ManifestLinkCheckItem){.path = path, .file = target1->file, .targetIdx = targetIdx});
+                lstSort(linkCheck->linkList, sortOrderAsc);
+
+                // Find the path in the sorted list
+                unsigned int linkIdx = lstFindIdx(linkCheck->linkList, &path);
+                ASSERT(linkIdx != LIST_NOT_FOUND);
+
+                // Check path links against other links (file links have already been checked)
+                if (target1->file == NULL)
+                {
+                    // Check the link destination path to be sure it is not a subpath of a prior link destination path
+                    for (unsigned int priorLinkIdx = linkIdx - 1; priorLinkIdx < linkIdx; priorLinkIdx--)
```

Comment:
I'm a little confused on how the test expression `priorLinkIdx < linkIdx` ever evaluates to false. If every `priorLink->file != NULL` we continue until when? Shouldn't this be `(int)priorLinkIdx >= 0`?

Reply:
We can make it `(int)priorLinkIdx >= 0` if you like, but this works as well because `(unsigned)0 - 1 == UINT_MAX`. I was hoping it would be more readable...

Reply:
we use the `(int)priorLinkIdx >= 0` construct in other places in the code so I'd rather be consistent...

Reply:
Sure thing, updated in d8e9d3db6ec.